uint8[]  ids
float64[]  present_angles
float64[]  goal_angles